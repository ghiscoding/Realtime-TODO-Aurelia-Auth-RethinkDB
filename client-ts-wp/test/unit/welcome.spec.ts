import {bootstrap} from 'aurelia-bootstrapper';
import {StageComponent} from 'aurelia-testing';
import {PLATFORM} from 'aurelia-pal';

describe('WelcomeComponent', () => {
  let component;

  beforeEach(async () => {
    component = StageComponent
      .withResources(PLATFORM.moduleName('welcome'))
      .inView('<welcome></welcome>');
    await component.create(bootstrap);
  });

  // only jest supports creating snapshot:
  if (jest) {
    it('should render correctly', () => {
      expect(document.body.outerHTML).toMatchSnapshot();
    });
  }

  it('should render first name', () => {
    const nameElement = document.querySelector('#fn') as HTMLInputElement;
    expect(nameElement.value).toBe('John');
  });

  afterEach(() => {
    component.dispose();
  });
});
