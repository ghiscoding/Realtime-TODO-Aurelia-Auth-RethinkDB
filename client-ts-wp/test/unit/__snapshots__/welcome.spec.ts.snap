// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeComponent should render correctly 1`] = `
"<body><div><welcome class=\\"au-target\\" au-target-id=\\"6\\">
  <section class=\\"au-animate\\">
    <h2>Welcome to the Aurelia Navigation App</h2>
    <form role=\\"form\\" submit.delegate=\\"submit()\\" class=\\"au-target\\" au-target-id=\\"2\\">
      <div class=\\"form-group\\">
        <label for=\\"fn\\">First Name</label>
        <input type=\\"text\\" value.bind=\\"firstName\\" class=\\"form-control au-target\\" id=\\"fn\\" placeholder=\\"first name\\" au-target-id=\\"3\\">
      </div>
      <div class=\\"form-group\\">
        <label for=\\"ln\\">Last Name</label>
        <input type=\\"text\\" value.bind=\\"lastName\\" class=\\"form-control au-target\\" id=\\"ln\\" placeholder=\\"last name\\" au-target-id=\\"4\\">
      </div>
      <div class=\\"form-group\\">
        <label>Full Name</label>
        <p class=\\"help-block\\">JOHN DOE</p>
      </div>
      <button type=\\"submit\\" class=\\"btn btn-default\\">Submit</button>
    </form>
  </section>
</welcome></div></body>"
`;
